client/entrypoints/express-checkout/utils/normalize.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Normalizes incoming cart total items for use as a displayItems with the Stripe api.
+ *
+ * @param {Array} displayItems Items to normalize.
+ *
+ * @return {Array} An array of PaymentItems
+ */
+export const normalizeLineItems = ( displayItems ) => {
+	return displayItems.map( ( displayItem ) => {
+		let amount = displayItem.value;
+		if ( displayItem.amount ) {
+			amount = displayItem.amount;
+		}
+		if ( displayItem.key === 'total_discount' ) {
+			amount = -amount;
+		}
+
+		return {
+			name: displayItem.label,
+			amount,
+		};
+	} );
+};
+
+/**
+ * Normalize order data from Stripe's object to the expected format for WC.
+ *
+ * @param {Object} event Stripe's event object.
+ * @param {string} paymentMethodId Stripe's payment method id.
+ *
+ * @return {Object} Order object in the format WooCommerce expects.
+ */
+export const normalizeOrderData = ( event, paymentMethodId ) => {
+	const name = event?.billingDetails?.name;
+	const email = event?.billingDetails?.email ?? '';
+	const billing = event?.billingDetails?.address ?? {};
+	const shipping = event?.shippingAddress ?? {};
+	const fraudPreventionTokenValue = window.wcStripeFraudPreventionToken ?? '';
+
+	const phone =
+		event?.billingDetails?.phone?.replace( /[() -]/g, '' ) ??
+		event?.payerPhone?.replace( /[() -]/g, '' ) ??
+		'';
+
+	return {
+		billing_first_name:
+			name?.split( ' ' )?.slice( 0, 1 )?.join( ' ' ) ?? '',
+		billing_last_name: name?.split( ' ' )?.slice( 1 )?.join( ' ' ) ?? '-',
+		billing_company: billing?.organization ?? '',
+		billing_email: email ?? event?.payerEmail ?? '',
+		billing_phone: phone,
+		billing_country: billing?.country ?? '',
+		billing_address_1: billing?.line1 ?? '',
+		billing_address_2: billing?.line2 ?? '',
+		billing_city: billing?.city ?? '',
+		billing_state: billing?.state ?? '',
+		billing_postcode: billing?.postal_code ?? '',
+		shipping_first_name:
+			shipping?.name?.split( ' ' )?.slice( 0, 1 )?.join( ' ' ) ?? '',
+		shipping_last_name:
+			shipping?.name?.split( ' ' )?.slice( 1 )?.join( ' ' ) ?? '',
+		shipping_company: shipping?.organization ?? '',
+		shipping_phone: phone,
+		shipping_country: shipping?.address?.country ?? '',
+		shipping_address_1: shipping?.address?.line1 ?? '',
+		shipping_address_2: shipping?.address?.line2 ?? '',
+		shipping_city: shipping?.address?.city ?? '',
+		shipping_state: shipping?.address?.state ?? '',
+		shipping_postcode: shipping?.address?.postal_code ?? '',
+		shipping_method: [ event?.shippingRate?.id ?? null ],
+		order_comments: '',
+		payment_method: 'woocommerce_payments',
+		ship_to_different_address: 1,
+		terms: 1,
+		'w-stripe-payment-method': paymentMethodId,
```

Comment:
```suggestion
		'wc-stripe-payment-method': paymentMethodId,
```

Reply:
Fixed in [0af7c3e](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3436/commits/0af7c3ee33a2b6aff8623927caf7870ae1fbf016)